### PR TITLE
added dockerfile with ubuntu and desert

### PR DESCRIPTION
**Pull request by sarafalleni05**

### CODE REVIEW

File: DESERT_Framework/Docker/ubuntu_dockerfile_with_desert/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,109 @@
+FROM ubuntu:jammy
+MAINTAINER Open Source Integrators <support@opensourceintegrators.com>
```
Comment (fedefava86) [inline DESERT_Framework/Docker/ubuntu_dockerfile_with_desert/Dockerfile:2]: who are they? Probably is worth indicating sth internal here, or if you want to, you can indicate your mail :)

File: DESERT_Framework/Docker/ubuntu_dockerfile_with_desert/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+# CPP deps
+RUN DEBIAN_FRONTEND=noninteractive \
+       apt-get install -qy \
+         build-essential \
+         ccache \
+         cmake \
+         libad9361-0 \
+         libasound2 \
+        #  libboost-date-time1.65.1 \
```
Comment (fedefava86) [inline DESERT_Framework/Docker/ubuntu_dockerfile_with_desert/Dockerfile:33]: I would remove all commented things, my 2 cents

**Review by fedefava86 — CHANGES_REQUESTED** [review]
Reallly helpful and good work indeed! Thanks @sarafalleni05 ! Few details to fix imho but we are already in a good shape

**Comment by sarafalleni05** [comment]
Hello,

I have addressed all the comments. Let me know if there is anything else.

Best,
Sara


Sara Falleni, (She/Her/Hers) PhD Candidate
Department of Electrical and Computer Engineering, Northeastern University
The Institute for the Wireless Internet of Things at Northeastern University
Email: ***@***.******@***.***>

On May 7, 2024, at 3:32 AM, Federico Favaro ***@***.******@***.***>> wrote:



@fedefava86 requested changes on this pull request.

Reallly helpful and good work indeed! Thanks @sarafalleni05<https://github.com/sarafalleni05> ! Few details to fix imho but we are already in a good shape

________________________________

In DESERT_Framework/Docker/ubuntu_dockerfile_with_desert/Dockerfile<https://github.com/signetlabdei/DESERT_Underwater/pull/19#discussion_r1591939100>:

> @@ -0,0 +1,109 @@
+FROM ubuntu:jammy
+MAINTAINER Open Source Integrators ***@***.******@***.***>>


who are they? Probably is worth indicating sth internal here, or if you want to, you can indicate your mail :)

________________________________

In DESERT_Framework/Docker/ubuntu_dockerfile_with_desert/Dockerfile<https://github.com/signetlabdei/DESERT_Underwater/pull/19#discussion_r1591939806>:

> +    rsync \
+    tar \
+    vim \
+    unzip \
+  && apt clean all \
+  && apt autoremove
+
+# CPP deps
+RUN DEBIAN_FRONTEND=noninteractive \
+       apt-get install -qy \
+         build-essential \
+         ccache \
+         cmake \
+         libad9361-0 \
+         libasound2 \
+        #  libboost-date-time1.65.1 \


I would remove all commented things, my 2 cents

________________________________

In DESERT_Framework/Docker/ubuntu_dockerfile_with_desert/Dockerfile<https://github.com/signetlabdei/DESERT_Underwater/pull/19#discussion_r1591941147>:

> +        #  pkg-config \
+         --no-install-recommends
+RUN DEBIAN_FRONTEND=noninteractive \
+       apt-get install -qy \
+       flex \
+       bison \
+       automake \
+       autoconf \
+       libtool
+
+
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -qy software-properties-common
+RUN DEBIAN_FRONTEND=noninteractive add-apt-repository -y ppa:git-core/ppa
+RUN DEBIAN_FRONTEND=noninteractive apt-get update && apt-get install -qy git
+
+# RUN mkdir /src && cd /src \


This has nothing to do with our stuff so i would remove all of this. I guess this is something belonging to the older dockerfile you used as starting point :)

________________________________

In DESERT_Framework/Docker/ubuntu_dockerfile_with_desert/Dockerfile<https://github.com/signetlabdei/DESERT_Underwater/pull/19#discussion_r1591948348>:

> +
+RUN DEBIAN_FRONTEND=noninteractive \
+        apt-get install -qy netcat \
+        && apt-get install -qy net-tools \
+        && apt-get install -qy iputils-ping \
+        && apt-get install -qy wget \
+        && apt-get install -qy vim \
+        && apt-get install -qy iproute2
+
+
+WORKDIR /tmp
+RUN git clone -b master https://github.com/signetlabdei/DESERT_Underwater.git \
+    && cd DESERT_Underwater/DESERT_Framework \
+    && ./install.sh --target LOCAL --with-woss --inst_mode development --addons ALL --dest_folder /tmp/DESERT_Underwater/DESERT_buildCopy_LOCAL
+
+RUN echo "DESERT is installed in the /tmp/ folder"


Consider that this is showing up once you build the container on your own, but we plan to ship also the pre-built image on docker.io<http://docker.io/>. This means that the final user will not see this. I was thinking about using WORKDIR in order to point directly the user to the sample folder, moreover we could use ENTRYPOINT in order to automatically source the needed file, or, as an alternative, we could write it directly in .bashrc since we do have only root user i guess inside the overlay fs of docker container.

**Review by fedefava86 — APPROVED** [review]

**Pull request merged**